[10.0] [ADD] sale_procurement_group_by_requested_date

AaronHForgeFlow opened the pull request:
Supersedes #541

serpentcs-dev1 committed:
sale_procurement_group_by_requested_date v9

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

nikul-serpentcs committed:
[MIG] sale_procurement_group_by_requested_date v10

AaronHForgeFlow committed:
[FIX]remove dependency to sale_sourced_by_line

[FIX]sale_procurement_group_by_requested_date does not depend on sale_sourced_by_line

AaronHForgeFlow commented:
Needs fixing

AaronHForgeFlow commented:
ping @mreficent

MiquelRForgeFlow commented:
If removing dependency, I think this needs a glue module.

AaronHForgeFlow commented:
sale_sourced_by_line is deprecated as long as we can add routes to the sale order lines, IMO no glue module is needed.

nikul-serpentcs commented on sale_procurement_group_by_requested_date/__manifest__.py:
```diff
@@ -0,0 +1,21 @@
+    "summary": "Groups pickings based on requested date of order line",
+    "version": "10.0.1.0.0",
+    "category": "Sales Management",
+    "website": "http://www.eficent.com",
```
Replace URL as per [this](https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py#L11)

nikul-serpentcs commented on sale_procurement_group_by_requested_date/models/sale_order.py:
```diff
@@ -0,0 +1,54 @@
+    def _prepare_order_line_procurement(self, group_id=False):
+        values = super(SaleOrderLine, self).\
+            _prepare_order_line_procurement(group_id=group_id)
+        self.ensure_one()
```
@aheficent no need, Remove it, Because already done [here](https://github.com/OCA/OCB/blob/10.0/addons/sale/models/sale.py#L638)

nikul-serpentcs commented on sale_procurement_group_by_requested_date/tests/test_sale_procurement_group_by_requested_date.py:
```diff
@@ -0,0 +1,114 @@
+        self.sale1.action_confirm()
+        req_date = fields.Date.to_string(self.dt1)
+        g_name = self.sale1.name + '/' + req_date
+        groups = self.env['procurement.group'].search([('name', '=', g_name)])
```
`self.env['procurement.group']` & `self.env['procurement.order']` use multiple time, add like global variable.

nikul-serpentcs reviewed (commented):
Minor Change

AaronHForgeFlow commented:
@nikul-serpentcs comments attended

MiquelRForgeFlow reviewed (approved):
LGTM

please squash

nikul-serpentcs reviewed (approved):
Code Review LGTM :+1:

AaronHForgeFlow commented:
It seems the tests for `sale_mrp_link` are failing here. I will check if it is due this module

MiquelRForgeFlow commented:
Have you checked?

AaronHForgeFlow commented:
> Have you checked?

Not yet

AaronHForgeFlow commented:
Could not reproduce on local. Rebasing.

AaronHForgeFlow commented:
Glue module is needed, will be proposed here.

AaronHForgeFlow committed:
[UPT]sale_procurement_group_by_line incompatible with sale_mrp_link. Sale as the other modules that are already excluded.

[ADD]sale_procurement_group_by_requested_date_sale_sourced_by_line. Glue module.

AaronHForgeFlow commented:
Of course the solution was to exclude this module to be installed together with sale_mrp_link. It happened the same to other modules that group procurements.
BTW it seems 10.0 branch is broken anyway. Edit: no, it is not. :smile:

tafaRU commented:
@aheficent, I think at the moment it's the only solution. See https://github.com/OCA/sale-workflow/issues/717#issuecomment-430231334 as further info.

AaronHForgeFlow commented:
@tafaRU Thanks, yes I think that is the only solution.

AaronHForgeFlow commented:
Added glue module for sale_sourced_by_line and sale_procurement_group_by_requested_date.  Sorry for the super large-ugly name.

AaronHForgeFlow commented:
It seems sale_delivery_split_date has higher priority key than the other modules that group procurements. As having all them installed together makes no sense, I exclude them to be installed together. Only sale_sourced_by_line and sale_procurement_group_by_requested_date is compatible using the glue module proposed here.

AaronHForgeFlow commented:
I seems there is a test failing on sale_triple_discount module. It is already happening in OCA/10.0 branch for this repo.

bjeficent reviewed (approved):
LGTM :+1:

AaronHForgeFlow commented:
This can be merged IMO

rafaelbn commented:
Please @aheficent This branch has conflicts that must be resolved

After this I will merge! Thanks

MiquelRForgeFlow commented:
@rafaelbn ready to merge. The travis red part is unrelated to this PR, is related to sale_triple_discount that is being fixed in another PR.

rafaelbn commented:
/ocabot merge
